step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/execution/IsolatedAutomationPackageRepository.java

```diff
@@ -177,6 +181,55 @@ public void cleanUpOutdatedResources() {
         log.info("Cleanup outdated automation packages finished. {} of {} packages have been removed", removed, foundResources.size());
     }
 
+    public void cleanUpMavenCache() {
+        log.info("Cleanup outdated artifacts in maven cache...");
+        String ttlString = ttlValueSupplier.get();
+
+        long ttlDurationMs = Long.parseLong(ttlString);
+        Long minExecutionTime = System.currentTimeMillis() - ttlDurationMs;
+
+        int removed = 0;
+        int artifactsCandidate = 0;
+        try {
+            Path mavenCachePath = Path.of("maven");
+            List<Path> mavenJarFiles = findOldJarFiles(mavenCachePath, minExecutionTime);
+            artifactsCandidate = mavenJarFiles.size();
+            for(Path jarPath: mavenJarFiles) {
+                File parentFile = jarPath.toFile().getParentFile();
+                if (parentFile != null) {
+                    boolean deleted = FileHelper.safeDeleteFolder(parentFile);;
+                    if (deleted) {
+                        removed++;
+                    }
```

Comment:
We should add a warning if it is not deleted

Reply:
done